gradle/build.go

```diff
@@ -79,12 +100,13 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		}
 	}
 
-	u, err := user.Current()
+	homeDir, err := b.HomeDirectoryResolver.Location()
 	if err != nil {
-		return libcnb.BuildResult{}, fmt.Errorf("unable to determine user home directory\n%w", err)
+		return libcnb.BuildResult{}, fmt.Errorf("home directory resolution failure\n%w", err)
```

Comment:
Yes, I changed it because I had to add a layer of indirection around fetching the user's home directory. This was causing issues with tests because it was using my actual home directory. Since the interface I added was a HomeDirectoryResolver, I wanted to just make it clear where the message was happening.

🤦 I'm embarrassed I didn't pick up on the pattern in the error messages, it's pretty obvious in hindsight. I will stick to that and just rename this one.